src/lwc/geBatchGiftEntryTable/geBatchGiftEntryTable.js

```diff
@@ -150,12 +158,14 @@ export default class GeBatchGiftEntryTable extends LightningElement {
             row[idProperty] === dataRow[idProperty]
         );
 
+        let rows = this.setDatatableRecordsForImperativeCall([dataRow]);
+
         if (existingRowIndex !== -1) {
             this.data.splice(existingRowIndex, 1, dataRow);
             this.data = [...this.data];
         } else {
-            this.data = [dataRow, ...this.data];
-            if (this.hasData == false) {
+            this.data = [...rows, ...this.data];
+            if (this.hasData === false) {
```

Comment:
`hasData` is undefined when the component is initialized.  With this change, should the field be updated to have a default value of `false`?

Reply:
done!

Reply:
👍🏽